src/neo4j_graphrag/neo4j_queries.py

```diff
@@ -171,6 +172,50 @@ def _get_hybrid_query(neo4j_version_is_5_23_or_above: bool) -> str:
     return call_prefix + query_body
 
 
+def _get_hybrid_query_linear(neo4j_version_is_5_23_or_above: bool, alpha: float) -> str:
+    """
+    Construct a Cypher query for hybrid search using a linear combination approach with an alpha parameter.
+
+    This query retrieves normalized scores from both the vector index and full-text index. It then
+    computes the final score as a weighted sum:
+
+    ```
+    final_score = alpha * (vector normalized score) + (1 - alpha) * (fulltext normalized score)
+    ```
+
+    If a node appears in only one index, the missing score is treated as 0.
+
+    Args:
+        neo4j_version_is_5_23_or_above (bool): Whether the Neo4j version is 5.23 or above; determines the call syntax.
+        alpha (float): Weight for the vector index normalized score. The full-text score is weighted as (1 - alpha).
+
+    Returns:
+        str: The constructed Cypher query string.
+    """
+    call_prefix = "CALL () { " if neo4j_version_is_5_23_or_above else "CALL { "
+
+    query_body = (
+        f"{NODE_VECTOR_INDEX_QUERY} "
+        "WITH collect({node: node, score: score}) AS nodes, max(score) AS vector_index_max_score "
+        "UNWIND nodes AS n "
+        "RETURN n.node AS node, (n.score / vector_index_max_score) AS score, 'vector' AS source "
```

Comment:
We could get rid of the source with something like:
```
WITH n.node AS node, (n.score / vector_index_max_score) AS score
RETURN score as rawScore, score * $alpha as score
```
? (just a proposal, seems to me we avoid an if/else and simplify the syntax but there might be drawbacks as well)

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/284/commits/98e8faf10e397b922c0d8f58cdf93fba73488702

I think that's way clearer, thanks!